easybuild/framework/easyconfig/tools.py

```diff
@@ -140,6 +141,225 @@ def find_resolved_modules(unprocessed, avail_modules, retain_all_deps=False):
 
     return ordered_ecs, new_unprocessed, new_avail_modules
 
+def toolchain_hierarchy_cache(func):
+    mydict = dict()
+    def wrapped_func(arg):
+        myarg = (arg['name'], arg['version'])
+        if myarg in mydict:
+            _log.debug("Using cache to return hierarchy for toolchain %s" % arg)
+            return mydict[myarg]
+        else:
+            new_val = func(arg)
+            mydict[myarg] = new_val
+            return new_val
+    return wrapped_func
+
+@toolchain_hierarchy_cache
+def get_toolchain_hierarchy(parent_toolchain):
```

Comment:
ah, wait, the latter would imply having easyconfigs around for all of them, so maybe not ;)